zio-json/shared/src/main/scala/zio/json/internal/readers.scala

```diff
@@ -51,6 +51,12 @@ private[zio] final class UnexpectedEnd
     )
     with NoStackTrace
 
+private[zio] final class RewindTwice
+    extends Exception(
+      "RecordingReader's rewind was called twice"
+    )
+    with NoStackTrace
```

Comment:
since this indicates a developer bug, rather than an input error, you might want to keep the stack trace here.